client/packages/cli_new/src/commands/register/gateway.ts

```diff
@@ -0,0 +1,255 @@
+import "@t3rn/types"
+import ora from "ora"
+import { Encodings } from "@t3rn/sdk"
+import { ApiPromise, WsProvider } from "@polkadot/api"
+import fetch from "node-fetch"
+import { createType } from "@t3rn/types"
+import {
+  T3rnPrimitivesGatewayVendor,
+  T3rnPrimitivesExecutionVendor,
+  T3rnAbiRecodeCodec,
+} from "@polkadot/types/lookup"
+import { Config, Gateway } from "@/schemas/setup.ts"
+import { colorLogMsg, log } from "@/utils/log.ts"
+import { createCircuitContext } from "@/utils/circuit.ts"
+
+export const spinner = ora()
+
+export const handleRegisterGateway = async (
+  config: Config,
+  gatewayId: string
+) => {
+  const foundGateway = config.gateways.find(
+    (g) => g.id.toLowerCase() === gatewayId.toLowerCase()
+  )
+
+  if (!foundGateway) {
+    log("ERROR", `Gateway ID ${gatewayId} not found in config file`)
+    process.exit(1)
+  }
+
+  spinner.text = `Registering ${foundGateway.name} gateway...`
+  spinner.start()
+  await registerGateway(foundGateway as Required<Gateway>)
+}
+
+const registerGateway = async (gatewayData: Required<Gateway>) => {
+  const { circuit, sdk } = await createCircuitContext()
+
+  const gatewayId = createType("[u8; 4]", gatewayData.id)
+  const tokenId = createType("[u8; 4]", gatewayData.tokenId)
+  const verificationVendor: T3rnPrimitivesGatewayVendor = createType(
+    "T3rnPrimitivesGatewayVendor",
+    gatewayData.registrationData.verificationVendor
+  )
+  const executionVendor: T3rnPrimitivesExecutionVendor = createType(
+    "T3rnPrimitivesExecutionVendor",
+    gatewayData.registrationData.executionVendor
+  )
+  const codec: T3rnAbiRecodeCodec = createType("T3rnAbiRecodeCodec", "Scale")
```

Comment:
This should be pulled from the config

Reply:
Fixed; refer to commit [999ce85](https://github.com/t3rn/t3rn/pull/881/commits/999ce858e91f835c9e783b500b5583cb563330ea).